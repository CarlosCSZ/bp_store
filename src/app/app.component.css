.main {
  width: 100%;
  height: 100%;
  display: flex;
  flex-direction: column;
  /* height: 100vh; */
}
.header {
  display: flex;
  background-color: white;
  padding: 20px 10px 10px;
  justify-content: center;
}
.header-img {
  width: 150px;
}

.body-route {
  height: 88%;
  background-color: var(--ligth-gray);
}
