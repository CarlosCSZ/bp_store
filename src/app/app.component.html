<main class="main">
  <div class="header">
    <img class="header-img" src="assets/img/logo_banco_pichincha_portada.jpg" alt="portada">
  </div>
  <div class="body-route">
    <router-outlet></router-outlet>
  </div>
</main>
