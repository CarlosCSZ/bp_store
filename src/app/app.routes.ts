import { Routes } from '@angular/router';
import { PresentationComponent } from './pages/presentation/presentation.component';
import { NewProductComponent } from './pages/new-product/new-product.component';
import { EditProductComponent } from './pages/edit-product/edit-product.component';

export const routes: Routes = [
  {
    path: 'home',
    component: PresentationComponent,
  },
  {
    path: 'new-product',
    component: NewProductComponent,
  },
  {
    path: 'edit-product',
    component: EditProductComponent,
  },
  {
    path: '',
    redirectTo: 'home',
    pathMatch: 'full'
  },
  {
    path: '**',
    redirectTo: 'home',
    pathMatch: 'full'
  }
];
