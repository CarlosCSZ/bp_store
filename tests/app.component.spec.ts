import { TestBed } from '@angular/core/testing';
import { AppComponent } from '../src/app/app.component';

describe('AppComponent', () => {
  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [AppComponent],
    }).compileComponents();
  });

  test('should create the app', () => {
    const fixture = TestBed.createComponent(AppComponent);
    const app = fixture.componentInstance;
    expect(app).toBeTruthy();
  });

  test(`should have the 'bp-products' title`, () => {
    const fixture = TestBed.createComponent(AppComponent);
    const app = fixture.componentInstance;
    expect(app.title).toEqual('bp-products');
  });

  test('should match the initial snapshot', () => {
    const fixture = TestBed.createComponent(AppComponent);
    fixture.detectChanges();
    const compiled = fixture.nativeElement as HTMLElement;
    expect(compiled).toMatchSnapshot();
  });
});
