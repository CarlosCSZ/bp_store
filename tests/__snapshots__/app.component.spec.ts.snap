// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should match the initial snapshot 1`] = `
<div
  id="root2"
>
  <main
    class="main"
  >
    <div
      class="header"
    >
      <img
        alt="portada"
        class="header-img"
        src="assets/img/logo_banco_pichincha_portada.jpg"
      />
    </div>
    <div
      class="body-route"
    >
      <router-outlet />
      
    </div>
  </main>
</div>
`;
